cli/set_option.ts

```diff
@@ -0,0 +1,40 @@
+import type {Arguments} from 'yargs';
+import {PackOptionKey} from './config_storage';
+import type {PackOptions} from './config_storage';
+import {TimerShimStrategy} from '../testing/compile';
+import {ensureUnreachable} from '..';
+import * as path from 'path';
+import {print} from '../testing/helpers';
+import {printAndExit} from '../testing/helpers';
+import {storePackOptions} from './config_storage';
+
+interface SetOptionArgs {
+  manifestFile: string;
+  option: string;
+  value: string;
+}
+
+export async function handleSetOption({manifestFile, option, value}: Arguments<SetOptionArgs>) {
+  const manifestDir = path.dirname(manifestFile);
+  const options = validateOption(option, value);
+  storePackOptions(manifestDir, options);
+  print('Option stored successfully.');
+}
+
+function validateOption(option: string, value: string): PackOptions {
+  const validOptions = Object.values(PackOptionKey) as string[];
+  if (!validOptions.includes(option)) {
+    return printAndExit(`Unsupported option "${option}". Value options are: ${validOptions.join(', ')}`);
+  }
+  const key = option as PackOptionKey;
+  switch (key) {
+    case PackOptionKey.timerStrategy:
+      const validValues = Object.values(TimerShimStrategy) as string[];
```

Comment:
totally optional, but with your pattern, I think you could do this in a generic way outside the switch: `const validValues = Object.values(PackOptions[key]) as string[]`. Even your printAndExit is nicely generic.

Reply:
Hmm good thought, not sure it works though since `PackOptions[key]` isn't a runtime list of valid values, but merely a TypeScript type.